src/components/Missions/index.jsx

```diff
@@ -1,7 +1,92 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchMissionsData, reserveMission, leaveMission } from '../../redux/missions/MissionSlice';
 
 function Missions() {
-  return <h1>this is my mission man</h1>;
+  const dispatch = useDispatch();
+  const missions = useSelector((state) => state.missions.missions);
+  const reservedMissions = useSelector((state) => state.missions.reservedMissions);
+
+  useEffect(() => {
+    if (missions.length === 0) {
+      dispatch(fetchMissionsData());
+    }
+  }, [dispatch, missions.length]);
+
+  const handleJoinMission = (missionId) => {
+    dispatch(reserveMission(missionId));
+  };
+
+  const handleLeaveMission = (missionId) => {
+    dispatch(leaveMission(missionId));
+  };
+
+  return (
+    <div className="hidden md:block">
```

Comment:
I did notice a small issue with the mobile version—it seems like the content isn’t displaying. I think it might be due to the hidden class on the main div, which is making the content visible only on desktop. If you could remove that class, it should help make the content appear correctly on mobile as well.